src/components/MapView/Layers/BoundaryLayer/index.tsx

```diff
@@ -40,12 +46,30 @@ function BoundaryLayer({ layer, before }: ComponentProps) {
     | undefined;
   const { data } = boundaryLayer || {};
 
+  const { i18n: i18nLocale } = useSafeTranslation();
+  const locationLevelNames = isEnglishLanguageSelected(i18nLocale)
+    ? layer.adminLevelNames
+    : layer.adminLevelLocalNames;
+
+  const isPrimaryLayer = isPrimaryBoundaryLayer(layer);
+
+  useEffect(() => {
+    if (!data || !isPrimaryLayer) {
+      return;
+    }
+
+    const boundaryRelationData = loadBoundaryRelations(
```

Comment:
You could run this once and use both adminLevelNames and adminLevelLocalNames values in your mapping so that this doesn't have to run when users switch language, wdyt?